service/email.go

```diff
@@ -0,0 +1,121 @@
+package service
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/hwsc-org/hwsc-user-svc/conf"
+	"io/ioutil"
+	"net/smtp"
+	"strings"
+	"text/template"
+)
+
+// Request holds transaction email data
+type emailRequest struct {
+	from         string
+	to           []string
+	subject      string
+	body         string
+	templateData map[string]string
+}
+
+const (
+	// MIME (Multipurpose Internet Mail Extension), extends the format of email
+	mime                = "MIME-version: 1.0;\nContent-Type: text/html; charset=\"UTF-8\";\n\n"
+	subjectVerifyEmail  = "Verify account for Humpback whale Social Call"
+	templateVerifyEmail = "verify_email.html"
+)
+
+func newEmailRequest(data map[string]string, to []string, from string, subject string) (*emailRequest, error) {
+	// note, data can be nil
+	if to == nil || from == "" || subject == "" {
+		return nil, errEmailRequestFieldsEmpty
+	}
+
```

Comment:
Emails can be checked, i know that is possible in Java, but not sure about in Golang. There is a package called https://github.com/badoux/checkmail

Reply:
I am not sure if we should let gmail handle the checking for us

Reply:
I can actually use this package to check email validity before inserting the new user into the database.

Do you think I will still need to check if email exist before sending any kind of email?

Reply:
And we don't have to use gmail, in fact using gmail with plainAuth is an issue because it forces me to enable "less secure apps" which isn't secure atm. But for testing purposes it serves well.